### ISSUE TITLE
Coachmark: React/WC/Figma/Docs design parity - dev updates

### ISSUE THREAD

**Issue by sunny-babbar** [carbon-design-system/ibm-products#9490]
Follow-on #9448 and reference #9100  

React / Web components review:
-------------------------------
### Coachmark tooltip
----

1. ### Missing caret

Coachmark tooltip
- React has a caret and Web components is missing the caret. Please add a caret to Web components. 

<img width="2386" height="1329" alt="Image" src="https://github.com/user-attachments/assets/a712fff3-8983-4bee-9af7-5bd10c63235f" />

2. ### Missing controls

Coachmark tooltip

React has more controls that Web components below are what is missing. 

React: 
-align
-caret **(Missing in Web components)**
-children **(Missing in Web components)**
-className **(Missing in Web components)**
-dropShadow
-floating
-highContrast
-onClose **(Missing in Web components)**
-open
-position
-selectorPrimaryFocus **(Missing in Web components)**
-triggerRef **(Missing in Web components)**

<img width="2553" height="1324" alt="Image" src="https://github.com/user-attachments/assets/4793bd44-1a35-4daf-88d6-7c3ddb54eb95" />

3. ### Figma / Coachmark docs

Figma's visuals and layout are looking much different compared to what is shown in React/Web components to code. Please update both React and Web components to look like the Figma example/[Coachmark docs](https://pages.github.ibm.com/carbon/ibm-products/components/onboarding/coachmark/usage/) below along with the same align and caret options. 

**The anatomy** 
Add the following
- [ ] Title
- [ ] Body content
- [ ] Media (optional)
- [ ] Dive deeper action (optional)
- [ ] Pagination (optional)
- [ ] Back button (optional)
- [ ] Primary action button
- [ ] Close icon
- [ ] Draggable icon (optional)

<img width="475" height="793" alt="Image" src="https://github.com/user-attachments/assets/232eaaf5-2f56-404b-a863-7ff8b9dacd23" />

<img width="1915" height="965" alt="Image" src="https://github.com/user-attachments/assets/aba0acec-6317-43be-9921-e0eff3a03510" />

**Position/Alignment** 
Add the following

Position:
- [ ] Top
- [ ] Bottom
- [ ] Left
- [ ] Right

Alignment:
- [ ] Center 
- [ ] Start
- [ ] End

Keep in mind this to build on what is already in React and Web components.
What already exists and need to add what is listed above. 
Suggestion: Add one control for position and another for alignment to match Figma to React/Web components.

**Missing:**
Position:
- Top
- Bottom
- Left
- Right

Alignment:
- Center 
- Start
- End

**As-is:**
-align
top
top-left
top-right
bottom
bottom-left
bottom-right
left
left-bottom **(Missing in Web components)**
left-top **(Missing in Web components)**
right
right-bottom **(Missing in Web components)**
right-top **(Missing in Web components)**
-caret **(Missing in Web components)**
-children **(Missing in Web components)**
-className **(Missing in Web components)**
-dropShadow
-floating
-highContrast
-onClose **(Missing in Web components)**
-open
-position
-selectorPrimaryFocus **(Missing in Web components)**
-triggerRef **(Missing in Web components)**

<img width="999" height="913" alt="Image" src="https://github.com/user-attachments/assets/cfd772e8-bc07-45c9-b7e5-d359995c5322" />
<img width="1003" height="920" alt="Image" src="https://github.com/user-attachments/assets/4748eba1-f919-4d22-828e-79f43c7f9001" />

<img width="656" height="316" alt="Image" src="https://github.com/user-attachments/assets/339c5a2f-2983-4869-96c8-06af09287ca3" />

**Note: Also consider the beacon to follow the position/alignment and caret**

<img width="462" height="751" alt="Image" src="https://github.com/user-attachments/assets/646177d9-1824-4e22-ac12-a64b5e11e2fd" />

**Note: Also consider all the steps above and apply it for the floating coachmark and stacked coachmark as well**

<img width="359" height="507" alt="Image" src="https://github.com/user-attachments/assets/b4a862f7-4e20-4ec7-b26b-d8264cc34acd" />

<img width="772" height="874" alt="Image" src="https://github.com/user-attachments/assets/3dfbf18b-31bb-490b-a37f-338bf1946191" />

4. ### Themes/ high contrast

In the Coachmark docs "Themes" are used and in React/Web components "highContrast" is used. What can be done here to make it consistent?

<img width="1917" height="969" alt="Image" src="https://github.com/user-attachments/assets/d7e4b5dd-9dc0-4009-997c-f4320bdf3c08" />

### Coachmark floating

----
5. ### Missing controls
Coachmark floating

React has more controls in the "Align" control dropdown than Web components below is what is missing.

align choose option dropdown

React:
-top
-top-left
-top-right
-bottom
-bottom-left
-bottom-right
-left
-left-bottom **(Missing in Web components)**
-left-top **(Missing in Web components)**
-right
-right-bottom **(Missing in Web components)**
-right-top **(Missing in Web components)**

<img width="1497" height="663" alt="Image" src="https://github.com/user-attachments/assets/1a7f7f04-48b2-40bc-91a3-7325c98abf18" />

6. ### Missing controls

Coachmark floating

React has more controls that Web components below are what is missing. 

React: 
-align
-caret **(Missing in Web components)**
-children **(Missing in Web components)**
-className **(Missing in Web components)**
-dropShadow
-floating
-highContrast
-onClose **(Missing in Web components)**
-open
-position
-selectorPrimaryFocus **(Missing in Web components)**
-triggerRef **(Missing in Web components)**

<img width="1919" height="897" alt="Image" src="https://github.com/user-attachments/assets/f2192c91-5202-4ce1-886b-96dd8919e548" />

7. ### Coachmark trigger ref

----

### Missing controls tab

Coachmark tigger ref

React has a tab for trigger ref, although Web components does not. Please add the trigger ref tab and all of the controls.

<img width="1917" height="911" alt="Image" src="https://github.com/user-attachments/assets/98679c32-d6a9-4ff6-9874-b0e4d679bd4f" />

### Additional updates needed
----

8. ### Dark theme link token (affecting the a11y)

- Web components link is not using dark theme link token and thats affecting the a11y

<img width="312" height="222" alt="Image" src="https://github.com/user-attachments/assets/daab04b9-3132-42c8-b68a-65ff5e443683" />
<img width="1909" height="920" alt="Image" src="https://github.com/user-attachments/assets/9cf76802-e439-4877-915e-daaefed7755f" />

- Same thing is happening with the ghost button here
<img width="1906" height="916" alt="Image" src="https://github.com/user-attachments/assets/70f60ece-773d-4548-967d-484637c8710b" />

9. ### Spacing

- Spacing between the description and the footer area should be 16px. This needs to be updated in the Figma component as well

<img width="1908" height="916" alt="Image" src="https://github.com/user-attachments/assets/2ff864d9-7978-458c-b548-24cc30555f12" />
<img width="630" height="750" alt="Image" src="https://github.com/user-attachments/assets/2c5ea3ed-197b-4f5c-8601-573b3c3f39bd" />

10. ### Button text should be left aligned

- Button text should be left aligned with title and description, and the idea icon looks off in code too. 

<img width="576" height="554" alt="Image" src="https://github.com/user-attachments/assets/79e09629-927f-4077-8ba6-5e7de058e511" />
<img width="1913" height="912" alt="Image" src="https://github.com/user-attachments/assets/96b92868-9778-4294-b870-fa5bc9fc0d7a" />

- Spacing

<img width="412" height="417" alt="Image" src="https://github.com/user-attachments/assets/0e83ffa9-9f36-4f3e-a260-e962995497ec" />

11. ### Stacked coachmark

- Some unexpected things are happening with the stacked coachmark. We are seeing a close button on both layers. There should only be a close button on the layer that is open and visible.

<img width="1915" height="926" alt="Image" src="https://github.com/user-attachments/assets/47192c3b-c844-4759-b64f-aa94c0f57dbd" />

<img width="391" height="417" alt="Image" src="https://github.com/user-attachments/assets/d75b69b7-e5cb-4eab-86bb-ebad975f40c4" />

12. ### Stroke around the container / caret

- On the react side looks like there's a stroke around the container which separates the caret from the container

<img width="1910" height="876" alt="Image" src="https://github.com/user-attachments/assets/07d537cd-54a8-43e1-8238-81ab7939b77f" />

Questions for dev
-------------

13. ### Animation (Dev please check on this and let us know)

- The animation also feels a bit choppy. The original motion design for the stacked coachmark had it sliding up from the bottom. We are wondering if there was a point where we decided to have it match tearsheet style? 
**Dev please let us know about this in a reply back**

**Comment by anamikaanu96** [comment]
1. Missing caret  - ✅

2. Missing controls - `selectorPrimaryFocus` :  ✅
                                    `caret` :  ✅
                                     `classname` and `children` - Not required : these are supported by default
                                     `onClose` - Replaced by the @c4p-coachmark-closed custom event
                                     `triggerRef` - Removed from React 

3. Figma / Coachmark docs -  I agree. At the same time, I'd like to point out that we took the reference from the deprecated version of Coachmark. Even in the stable version, it uses the same Tooltip example that I'm currently using. Since this is the new composable version of Coachmark, adopters can definitely add their own custom fields as needed.

Also, the images you mentioned are actually part of the Coachmark Overlay Elements which is a pattern built using the Coachmark component.

Missing controls : Position and Alignment - The alignment prop is responsible for both the position and alignment. I'm using the Popover alignment values, which is the same approach used in the Carbon Popover as well.

4. Themes/ high contrast - Both are consistent. In your image I see the highcontrast prop is set to `false`
     Please check again -  [React](https://carbon-design-system.github.io/ibm-products/staging/react/?path=/story/preview-onboarding-coachmark--tooltip) and [Web component](https://ibm-products-web-components.carbondesignsystem.com/?path=/story/components-onboarding-coachmark--tooltip)

5. Coachmark floating : Missing controls (Align) ✅

6. Coachmark floating : Missing controls - `selectorPrimaryFocus` : ✅                    
                                                                        `caret` : ✅
                                                                         `classname` and `children` - Not required : these are supported by default
                                                                         `onClose` - Replaced by the `@c4p-coachmark-closed` custom event
                                                                         `triggerRef` - Removed from React 
     
7. Coachmark trigger ref - Removed from React

8. Dark theme link token - Already raised an [issue](https://github.com/carbon-design-system/carbon/issues/21438) for this in Core Carbon

9. Spacing - **To be done**

10. Button text should be left aligned - **To be done**

11. Stacked coachmark - Although we referred to the deprecated version of Coachmark, even in the stable version it uses the same  example that I'm currently using - with close icon in parent and child.

12. Stroke around the container / caret - Already fixed

13. Animation - **To be done** 
      I'll update the animation to make it smoother and align it more closely with the deprecated stacked Coachmark animation, where      
      it slides up from the bottom.
      Regarding matching the Tearsheet animation, I don't think that's a requirement. Since this is a pattern, adopters have the 
      flexibility to implement the animation that best fits their use case

**Opened a separate ticket for unresolved issues ( style and animation issue , issue number : 9, 10, 13)**
